bots/sg_modify_scope_by_port.py

```diff
@@ -0,0 +1,187 @@
+"""
+##sg_modify_scope_by_port
+# What it does: modify Security Group's rules scope by a given port , new and old scope(optional).
+# Direction can be : inbound or outbound
+# Usage: AUTO: sg_modify_scope_by_port <port> <change_scope_from|*> <change_scope_to> <direction>
+       - When '*' set for replacing any rule with the specific port
+# Examples:
+        AUTO: sg_modify_scope_by_port 22 0.0.0.0/0 10.0.0.0/24 inbound
+        AUTO: sg_modify_scope_by_port 22 * 10.0.0.0/24 inbound
+#Notes:
+    -  if the port is in a rule's port range, the bot will change the rule's ip to desire ip , to avoid that
+      specify existing rule's scope instead of using '*'
+    - to split the rule around the port you can use the bot : #sg_single_rule_delete
+
+#Limitations: IPv6 is not supported yet
+
+"""
+import re
+
+PORT_TO = 'portTo'
+PORT_FROM = 'port'
+PROTOCOL = 'protocol'
+SCOPE = 'scope'
+ALL_TRAFFIC_PORT = 0
+ALL_TRAFFIC_PROTOCOL = '-1'
+
+"""
+checks if a rule exists in a security group , returns false/true 
+"""
+
+
+def is_rule_exists_in_sg(sg, rule, direction, scope):
```

Comment:
Is here you verify the modification?

Reply:
No, it's possible that the current modification will create a rule that already exists when using '*'(any) in the change_scope_from parameter. 
what happens is the bot changes by port no matter what the scope is, it is possible that there are a few rules with the same port and different scopes. after changing all those rule's scopes, duplicates will be created. when I call Boto3's creation function without checking duplicates with this function an Exception will raise.